File: samples/dart/06cat/cat.dart
import "dart:io";
import "dart:async";
import "dart:convert";


void main(List<String> arguments) {
    if (arguments.length > 1) {
        stderr.write("Error: Usage dart cat.dart [<file>]\n");
        exit(1);
    }

    Stream<List<int>> inputStream = stdin;
    if (arguments.length == 1) {
        var inFile = new File(arguments[0]);
        inputStream = inFile.openRead();
    }

    inputStream
        .transform(utf8.decoder)
        .transform(new LineSplitter())
        .listen((String _line) {
                print(_line);
            },
            onDone: () {},
            onError: (e) { stderr.write(e.toString()); });
}

File: samples/python/06cat/cat.py
import sys

if __name__ == '__main__':
    if len(sys.argv) > 2:
        print(f"Error: Usage {sys.argv[0]} [<file>].", file=sys.stderr)
        exit(1)

    infile = open(sys.argv[1],'r') if len(sys.argv) > 1 else sys.stdin

    line = infile.read(100)
    while line:
        sys.stdout.write(line)
        line = infile.read(100)